### PR TITLE
DBZ-156 Updates EmbeddedEngine to better handle exceptional cases and provide more feedback during startup

### PR DESCRIPTION
**Pull request by hchiorean**
It also updates `EmbeddedEngine` to use the Kafka commit callbacks introduced after 0.10 and updates `AbstractConnectorTest` to better synchronize with the embedded engine

### CODE REVIEW

File: debezium-embedded/src/main/java/io/debezium/embedded/EmbeddedEngine.java [hunk 1]
```diff
@@ -606,44 +675,49 @@ public void raiseError(Exception e) {
...
                 } catch (Throwable t) {
                     fail("Error while trying to run connector class '" + connectorClassName + "'", t);
-                    return;
                 } finally {
                     // Close the offset storage and finally the connector ...
                     try {
                         offsetStore.stop();
                     } finally {
                         connector.stop();
+                        connectorCallback.ifPresent(ConnectorCallback::connectorStopped);
```
Comment (rhauch) [inline debezium-embedded/src/main/java/io/debezium/embedded/EmbeddedEngine.java]: Should we handle the case when `connector.stop()` throws an error? Should line #684 be in a finally block?

Comment (hchiorean) [inline debezium-embedded/src/main/java/io/debezium/embedded/EmbeddedEngine.java]: don't know....the current callbacks only get called for non-exceptional cases
Currently if `connector.stop` or `task.stop` fails, no `ConnectorCallback` methods get called. If we want to handle these cases as well, then we could add an optional `Throwable` parameter to the callbacks for example.
That being said, if we do that we should also do the same for the `start` methods (which can also fail). 
So I'm not convinced; maybe the current `CompletionCallback` is enough to handle all exceptional cases.

File: debezium-embedded/src/test/java/io/debezium/embedded/AbstractConnectorTest.java [hunk 2]
```diff
@@ -279,6 +293,16 @@ protected void start(Class<? extends SourceConnector> connectorClass, Configurat
             LoggingContext.forConnector(getClass().getSimpleName(), "", "engine");
             engine.run();
         });
+        try {
+            if (!latch.await(10, TimeUnit.SECONDS)) {
+                // maybe it takes more time to start up, so just log a warning and continue
+                logger.warn("The engine startup has not completed in the expected amount of time;");
```
Comment (rhauch) [inline debezium-embedded/src/test/java/io/debezium/embedded/AbstractConnectorTest.java]: May want to say "The connector did not finish starting its task(s) or complete in the expected amount of time."

File: debezium-embedded/src/main/java/io/debezium/embedded/EmbeddedEngine.java (hunk 1 above)
Comment (rhauch) [inline debezium-embedded/src/main/java/io/debezium/embedded/EmbeddedEngine.java]: Okay, that makes sense.

File: debezium-embedded/src/test/java/io/debezium/embedded/AbstractConnectorTest.java (hunk 2 above)
Comment (hchiorean) [inline debezium-embedded/src/test/java/io/debezium/embedded/AbstractConnectorTest.java]: ok

**Comment by hchiorean** [comment]
amended the commit and changed the log warning message as per the comment above

**Pull request merged**